### PR TITLE
cohortManger and extractRadiomics app

### PR DESCRIPTION
**Pull request by aauker**

### CODE REVIEW

File: data_processing/cohortManager/cohortManager.py [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+        scan_dict = rec.data()['scan']
+        scan_id   = rec.data()['id(scan)']
+        scan_dict['id']   = scan_id
+        scan_dict['data'] = requests.get(f'http://{HOST}:5004/mind/api/v1/listData/{scan_id}').json()
+        all_scan.append(scan_dict)
+    return jsonify(all_scan)
+
+# List all the scans
+@app.route('/mind/api/v1/listData/<id>', methods=['GET'])
+def listData(id):
```
Comment (doori) [inline data_processing/cohortManager/cohortManager.py:75]: it'd be good to make the type of id more specific in the variable name and/or in the comment.

File: data_processing/cohortManager/cohortManager.py [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+# ==================================================================================================
+# Routes to add manage cohort/study structure (add/remove patient, add cases, add scans)
+# ==================================================================================================
+
+# Add (xnat) patient
+@app.route('/mind/api/v1/addXnatPatient/<cohort_id>/<patient_id>', methods=['GET'])
+def addPatient(cohort_id, patient_id):
+    res = conn.query(f"""MATCH (n) WHERE n.PatientID='{patient_id}' RETURN n""")
+    if len(res) == 0:
+        return ("No patients found")
```
Comment (doori) [inline data_processing/cohortManager/cohortManager.py:96]: No patients found with patient id: {patient_id}

File: data_processing/cohortManager/cohortManager.py [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+        conn.query(f"""MATCH (n) WHERE n.PatientID='{patient_id}' MERGE (m:patient{{name:'{patient_id}', cohort:'{cohort_id}'}}) SET m.active=True  MERGE (n)-[:PROXY]-(m)""")
+        return (f"Added {patient_id} to {cohort_id}")
+    else:
+        return ("Malformed patient id returned multiple types")
+
+# Remove (deactive) patient
+@app.route('/mind/api/v1/removePatient/<cohort_id>/<patient_id>', methods=['GET'])
+def removePatient(cohort_id, patient_id):
+    res = conn.query(f"""MATCH (m:patient{{name:'{patient_id}', cohort:'{cohort_id}'}}) SET m.active=False RETURN m""")
+    return ("Deleted {} patients from cohort".format(len(res)))
```
Comment (doori) [inline data_processing/cohortManager/cohortManager.py:107]: Curious, do we expect more than one patient to be deleted here?

File: data_processing/extractRadiomics/extractRadiomics.py [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+
+    with lock:
+        if not method_id in METHODS.keys():
+            dataset_id = f"RAD_{method_id}"
+            METHODS[method_id] = {}
+            METHODS[method_id]['config']      = request.json
+            METHODS[method_id]['cohort_id']   = cohort_id
+            METHODS[method_id]['dataset_id']  = dataset_id
+            METHODS[method_id]['streams_dir'] = os.path.join("/gpfs/mskmindhdp_emc/data/", cohort_id, "streams",  dataset_id)
+            METHODS[method_id]['dataset_dir'] = os.path.join("/gpfs/mskmindhdp_emc/data/", cohort_id, "tables", dataset_id)
```
Comment (doori) [inline data_processing/extractRadiomics/extractRadiomics.py:84]: let's move the hardcoded paths to some configs.

**Comment by doori** [comment]
Excellent work! some thoughts..
- we should have a discussion with the team on how we are using the configuration and env vars.
- please document the new graph nodes/properties.
- are the swagger documentation available for these endpoints?
- it would be good to think about what data goes into delta tables/ graphs or both.

File: data_processing/cohortManager/cohortManager.py [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+import itk
+
+from radiomics import featureextractor  # This module is used for interaction with pyradiomics
+import threading
+
+app    = Flask(__name__)
+logger = init_logger("flask-mind-server.log")
+cfg    = ConfigSet(name=const.APP_CFG,  config_file="config.yaml")
+spark  = SparkConfig().spark_session(const.APP_CFG, "data_processing.radiology.api.5004")
+conn   = Neo4jConnection(uri=os.environ["GRAPH_URI"], user="neo4j", pwd="password")
```
Comment (doori) [inline data_processing/cohortManager/cohortManager.py:23]: user/password should go in config/env vars

**Review by doori — APPROVED** [review]

**Comment by aauker** [comment]
Thanks Doori, I considered all your comments and prepared some TODOs come the new year.  Happy holidays!! ^^

**Pull request merged**